### ISSUE TITLE
Missing shell.nix

### ISSUE THREAD

**Issue by idontgetoutmuch** [acowley/Frames#99]
~/.local/bin/stack --nix build
error: getting status of 'Frames/shell.nix': No such file or directory

**Comment by teto** [comment]
regarding the shell.nix, it assumes that you've cloned Vynil which is not practical:
`vinyl = pkgs.haskell.lib.dontBenchmark (super.callPackage ~/Projects/Vinyl {});`

**Comment by acowley** [comment]
Yes, it's not used to build releases as of yet and only for the sake of development; anyone wanting to make use of it edits it to suit their setup.

If there is a desire for a more portable version, I'd like to figure out how to make things automatically work with `direnv`. We've done a `Vinyl` git submodule before, so that's an option again, though I felt that was clunkier than just having a side-by-side checkout with `nix` pointing to that. So what is needed is a setup where we can have a `shell.nix` for people sure they won't need to touch `Vinyl`, and one for folks who think they might, such that `direnv` does what is wanted. I imagine this is possible, but didn't figure it out with a quick look at how [nix-direnv](https://github.com/nix-community/nix-direnv/blob/master/direnvrc) is put together. I think [this](https://github.com/nix-community/nix-direnv/blob/master/direnvrc#L38) is the relevant line of `nix-direnv` that ideally we could instrument with a parameter or something in `.envrc` to control which variant of the `shell.nix` control flow is taken.
